encodings/fastlanes/src/for/pipeline.rs

```diff
@@ -178,11 +188,17 @@ where
     T: NativePType + Element + WrappingAdd,
     E: NativePType + Element,
 {
-    fn step(&mut self, ctx: &KernelContext, out: &mut ViewMut) -> VortexResult<()> {
+    fn step(
```

Comment:
shall we replace this with a arithmetic kernel?

Reply:
Eventually yes